### ISSUE TITLE
I'm not able to login

### ISSUE THREAD

**Issue by simonecogno** [jasonjoh/node-tutorial#27]
I tried to sign in but after I'm redirectet to microsoft page then I'm redirected back to the application but it does not work and a receive an error in the code.

These are the logs
```
GET /authorize?code=xxxxxx&session_state=xxxxx 302 240.960 ms - 46
Generated auth url: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?response_type=code&client_id=xxxxx&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauthorize&scope=openid%20profile%20offline_access%20User.Read%20Mail.Read%20Calendars.Read%20Contacts.Read
GET / - - ms - -
Error: /Users/xxxx/WebstormProject/node-tutorial/views/error.hbs: Can't set headers after they are sent.
    at validateHeader (_http_outgoing.js:491:11)
    at ServerResponse.setHeader (_http_outgoing.js:498:3)
    at ServerResponse.header (/Users/xxx/WebstormProject/node-tutorial/node_modules/express/lib/response.js:730:10)
    at ServerResponse.send (/Users/xxx/WebstormProject/node-tutorial/node_modules/express/lib/response.js:170:12)
    at done (/Users/xxx/WebstormProject/node-tutorial/node_modules/express/lib/response.js:967:10)
    at /Users/xxx/WebstormProject/node-tutorial/node_modules/hbs/lib/hbs.js:93:9
    at Object.done (/Users/xxx/WebstormProject/node-tutorial/node_modules/hbs/lib/async.js:74:20)
    at /Users/xxx/WebstormProject/node-tutorial/node_modules/hbs/lib/hbs.js:88:18
    at /Users/xxx/WebstormProject/node-tutorial/node_modules/hbs/lib/hbs.js:69:11
    at Object.done (/Users/xxx/WebstormProject/node-tutorial/node_modules/hbs/lib/async.js:74:20)
    at /Users/xxx/WebstormProject/node-tutorial/node_modules/hbs/lib/hbs.js:64:20
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:511:3)

```

Any idea why? I've followed the instruction at the letter.

Branch: master

**Comment by jasonjoh** [comment]
Well it looks like you got an error (it's trying to render error.hbs), it would be helpful to figure out what that error actually is. Can you hook up a debugger? Relevant code is probably the `router.get` calls in authorize.js.

**Comment by monikagupta78** [comment]
I am facing same issue, any solution you found?

**Comment by shouryalala** [comment]
No fix for this yet?
This issue has made all REST apis unusable for me as i'm unable to fetch an access token.

**Comment by jasonjoh** [comment]
The fix in #31 should clear this up, or at least give you more info on what's failing.
